### PR TITLE
doc: Add documentation details on role valid values and warning on application deletion

### PR DESCRIPTION
**Pull request by jasonouellet**
- Add details about possible values for role variables
- Add warning about the impact of deleting an application when changing the default environment after creation.

### CODE REVIEW

File: docs/resources/application_user.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ resource "humanitec_application_user" "another_owner" {
 ### Required
 
 - `app_id` (String) The Application ID.
-- `role` (String) The role that this user holds
+- `role` (String) The role that this user holds. Could be `viewer`, `developer` or `owner`.
```
Comment (johanneswuerbach) [inline docs/resources/application_user.md:29]: The docs are auto-generated from markdown descriptions inside the code, could you update https://github.com/humanitec/terraform-provider-humanitec/blob/main/internal/provider/resource_application_user.go#L82 (and the other places you touched) and run `go generate ./...` afterwards?

**Review by johanneswuerbach — CHANGES_REQUESTED** [review]
Thank you for your contribution @jasonouellet, but there are some changes that would be required.

**Review by johanneswuerbach — APPROVED** [review]

File: docs/resources/user.md [hunk 2]
```diff
@@ -18,8 +18,8 @@ An entity or individual who has access to the Humanitec platform.
 ### Required
 
 - `name` (String) The name the user goes by.
-- `role` (String) The role that the service user should have on the organization it is created in.
-- `type` (String) The type of the account. Could be user, service or system.
+- `role` (String) The role that the service user should have on the organization it is created in. Could be member`, `artefactcontributor`, `manager` or `administrator`.
```
Comment (mathieu-benoit) [inline docs/resources/user.md:21]: @mateuszjenek @johanneswuerbach, just to confirm here, `artefactcontributor` or `artefactContributor`?

Comment (johanneswuerbach) [reply docs/resources/user.md:21]: Good catch 🤕 https://github.com/humanitec/terraform-provider-humanitec/pull/82

Comment (mateuszjenek) [reply docs/resources/user.md:21]: Should be `artefactContributor`

**Pull request merged**